controllers/helmrepository_controller_test.go

```diff
@@ -291,11 +292,12 @@ func TestHelmRepositoryReconciler_reconcileSource(t *testing.T) {
 		protocol         string
 		server           options
 		secret           *corev1.Secret
-		beforeFunc       func(t *WithT, obj *sourcev1.HelmRepository, checksum string)
+		beforeFunc       func(t *WithT, memCache *cache.Cache, obj *sourcev1.HelmRepository, checksum string)
 		afterFunc        func(t *WithT, obj *sourcev1.HelmRepository, artifact sourcev1.Artifact, chartRepo repository.ChartRepository)
 		want             sreconcile.Result
 		wantErr          bool
 		assertConditions []metav1.Condition
+		cache            *cache.Cache
```

Comment:
can we just declare a `testCache` variable at the top of the function and pass it to the `beforeFunc` instead of having a dedicated field?

Reply:
implemented with the newly added global var